wsf_scraping/pipelines.py

```diff
@@ -25,63 +26,70 @@ def __init__(self):
-        pdf_path = ''.join(['./results/pdfs/', spider.name, '/'])
+        pdf_result_path = os.path.join(
+            os.path.curdir,
+            'results',
+            'pdfs',
+            spider_name,
+            '/'
+        )
 
         # Convert PDF content to text format
-        f = open('/tmp/' + item['pdf'], 'rb')
+        with open(base_pdf_path, 'rb') as f:
+            if download_only:
+                os.rename(
+                    ''.join(['/tmp/', item['pdf']]),
+                    ''.join([pdf_result_path, item['pdf']])
+                )
+                return item
 
-        if download_only:
-            os.rename(
-                ''.join(['/tmp/', item['pdf']]),
-                ''.join([pdf_path, item['pdf']])
+            self.logger.info(
+                'Processing: %s (%s)',
+                item['pdf'],
+                self.settings['PARSING_METHOD'],
             )
-            return item
 
-        self.logger.info('Processing: %s (%s)' % (
-            item['pdf'],
-            self.settings['PARSING_METHOD'],
-        ))
+            if self.settings['PARSING_METHOD'] == 'pdftotext':
+                pdf_file = parse_pdf_document_pdftxt(f)
+            else:
+                pdf_file = parse_pdf_document(f)
 
-        # Parse using pdftotext only if specified. Leads to more false positive
-        if self.settings['PARSING_METHOD'] == 'pdftotext':
-            pdf_file = parse_pdf_document_pdftxt(f)
-        else:
-            pdf_file = parse_pdf_document(f)
+            for keyword in self.section_keywords:
+                # Fetch references or other keyworded list
+                section = grab_section(pdf_file, keyword)
+
+                # Add references and PDF name to JSON returned file
+                # If no section matchs, leave the attribute undefined
+                if section:
+                    item['sections'][keyword.title()] = section
 
-        for keyword in self.section_keywords:
             # Fetch references or other keyworded list
-            section = grab_section(pdf_file, keyword)
+            keyword_dict = pdf_file.get_lines_by_keywords(
+                self.keywords,
+                self.settings['KEYWORDS_CONTEXT']
+            )
 
             # Add references and PDF name to JSON returned file
             # If no section matchs, leave the attribute undefined
-            if section:
-                item['sections'][keyword.title()] = section
-
-        # Fetch references or other keyworded list
-        keyword_dict = pdf_file.get_lines_by_keywords(
-            self.keywords,
-            self.settings['KEYWORDS_CONTEXT']
-        )
-
-        # Add references and PDF name to JSON returned file
-        # If no section matchs, leave the attribute undefined
-        if keyword_dict:
-            item['keywords'] = keyword_dict
-
-        f.close()
+            if keyword_dict:
+                item['keywords'] = keyword_dict
```

Comment:
Git has not picked up the diff in a nice way so that bit is hard to review. Looks ok.